docs/high-availability/provision.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DISCOVERY=$1
+IP=`ip -o addr | grep 192 | cut -d' ' -f 7 | cut -d'/' -f 1` 
+
+function initialize {
+    sudo systemctl stop firewalld
+    sudo setenforce 0
+    yum install -y docker
+    
+    ### Important : The kube version MUST match the containers in the manifests.
+    ### Otherwise lots of API errors.
+    sudo yum install -y http://cbs.centos.org/kojifiles/packages/kubernetes/0.17.1/3.el7/x86_64/kubernetes-node-0.17.1-3.el7.x86_64.rpm
+    sudo mkdir -p -m 777 /etc/kubernetes/manifests
+    sudo systemctl disable iptables-services firewalld
+    sudo systemctl stop iptables-services firewalld
+    ## yum install -y flannel ; 
+    ## service flanneld start
+    ### just to make it easy to hack around as non root user
+    sudo groupadd docker
+    sudo gpasswd -a vagrant docker
+    sudo service docker restart
+}
+
+function start_kubelet {
+    sudo systemctl enable docker   
+    ### We need a custom unit file with the --config/ option
+    cp /vagrant/etc_kubernetes_kubelet /etc/kubernetes/kubelet
+    sudo systemctl enable kubelet
+    ### Not sure why, but this restart is required?
+    sleep 2
+    sudo systemctl restart kubelet
+} 
+
+### Not the best idea if using flannel.  Because of the circular dependency.
+function write_etcd_manifest {
+
+        ### I know this looks fancy, but 
+        ### Basically, this is just setting up ETCD config file w/ IP Addresses
+        DISCOVERYESC="`echo "$DISCOVERY" | sed -e 's/[\/&]/\\\&/g'`"
+        sudo cat /vagrant/etcd.manifest | \
+        sed "s/DISCOVERY_TOKEN/$DISCOVERYESC/" | \
+        sed "s/NODE_NAME/`hostname`/" | \
+        sed "s/NODE_IP/$IP/" > /etc/kubernetes/manifests/etcd.manifest
```

Comment:
if pipelined commands can't fit on one line, they should appear once per line and be indented. the chaining keyword (|,&&,||,>) should also be a the begining of the line.
```
sudo cat /vagrant/etcd.manifest \
        | sed "s/DISCOVERY_TOKEN/$DISCOVERYESC/" \
        | sed "s/NODE_NAME/`hostname`/" \
        | sed "s/NODE_IP/$IP/" \
        > /etc/kubernetes/manifests/etcd.manifest